Makefile

```diff
@@ -16,31 +16,30 @@
 
 # CONFIG
 
-# Specify the device to build for
-devicefamily = stm32f0xx
-
-# Specify the mainprojectfile you want to build
-mainprojectfile = projects/main.c 
+# Specify the directory with the rule.mk file for the project you want to build 
+RULES_DIR := projects/test_project
+include $(RULES_DIR)/rules.mk
+MAIN_FILE = $(RULES_DIR)/$(MAIN)
 
 # compile directory
-bindir = bin
+BIN_DIR := bin
 
 # location of OpenOCD board .cfg files (only triggered if you use 'make program' explicitly)
-OPENOCD_BOARD_DIR=/usr/share/openocd/scripts/board
+OPENOCD_BOARD_DIR := /usr/share/openocd/scripts/board
 
+# Please don't touch anything below this line
 ###################################################################################################
 
 # AUTOMATED ACTIONS
 
-# name of generated *.elf files 
-mainprojectname = $(basename $(notdir $(mainprojectfile)))
+# name of generated *.elf file 
+MAIN_PROJECT := $(basename $(notdir $(MAIN_FILE)))
 
 # location of the libraries
-lib_dir = libraries/$(devicefamily)
-devdir = device/$(devicefamily)
-mslibdir = libraries/ms-lib
+LIB_DIR := libraries/$(DEVICE_FAMILY)
+DEV_DIR := device/$(DEVICE_FAMILY)
```

Comment:
Ideally, you wouldn't need to list every single library that the project might need. That should be in the project's makefile.